Tests/Marketplace/zip_packs.py

```diff
@@ -198,9 +247,9 @@ def get_latest_pack_zip_from_blob(pack, blobs):
         blob: The zip blob of the pack with the latest version.
     """
     blob = None
-    blobs = [b for b in blobs if os.path.splitext(os.path.basename(b.name))[0] == pack and b.name.endswith('.zip')]
+    blobs = [b for b in blobs if os.path.splitext(os.path.basename(b))[0] == pack and b.endswith('.zip')]
     if blobs:
-        blobs = sorted(blobs, key=lambda b: LooseVersion(os.path.basename(os.path.dirname(b.name))), reverse=True)
+        blobs = sorted(blobs, key=lambda b: LooseVersion(os.path.basename(os.path.dirname(b))), reverse=True)
```

Comment:
You're going over the list twice, you can do all the logic with one pass.

Reply:
This function logic was added in an earlier version, I just removed the attribute access to match the changes I made. Should I refactor this code anyway?

Reply:
I believe so since you're trying to optimize the functionality of the step.

Reply:
I read the code again and I can't see where it goes over the list twice. It seems that first it creates a list of only zip files and then it sorts that list. 
I'd be happy if you could please explain where you see an extra traversal on the list.

Reply:
Fixed it.